bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/asset/AssetIngestor.java

```diff
@@ -45,39 +46,63 @@
 import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.EnumMap;
 import java.util.List;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.concurrent.atomic.AtomicLong;
 
 public abstract class AssetIngestor extends ProcessDefinition {
-    private final MimeTypeService mimetypeService;
+    public static final String ALL_ASSETS = "All Assets";
+
+    protected final transient MimeTypeService mimetypeService;
 
     @SuppressWarnings("squid:S00115")
-    public enum AssetAction {
+    public static enum AssetAction {
         skip, version, replace
     }
 
     public AssetIngestor(MimeTypeService mimeTypeService) {
         this.mimetypeService = mimeTypeService;
     }
 
+    @FormField(
+            name = "Dry run",
+            description = "If checked, no import happens.  Useful for data validation",
+            component = CheckboxComponent.class
+    )
+    boolean dryRunMode = false;
+
+    @FormField(
+            name = "Detailed report",
+            description = "If checked, information about every asset is recorded",
+            component = CheckboxComponent.class,
+            options = "checked"
+    )
+    boolean detailedReport = true;
+    
+    @FormField(
+            name = "Inhibit workflow",
+            description = "If checked, disables asset processing workflow",
+            component = CheckboxComponent.class,
+            options = "checked"
+    )
+    boolean inhibitWorkflow = true;
+
     @FormField(
             name = "Target JCR Folder",
-            description = "Base folder for ingestion",
+            description = "Prepended to target path if it does not begin with /content",
             hint = "/content/dam",
             component = PathfieldComponent.FolderSelectComponent.class,
             required = true,
             options = {"default=/content/dam", "base=/content/dam"}
     )
-    String jcrBasePath;
+    String jcrBasePath = "/content/dam";
     @FormField(
             name = "Ignore folders",
             description = "List of folder names to be ignored",
             hint = "tmp,.DS_STORE",
             options = {"default=tmp,ds_store,.ds_store,.thumbs,.appledouble"}
     )
-    String ignoreFolders;
+    String ignoreFolders = "tmp,ds_store,.ds_store,.thumbs,.appledouble";
```

Comment:
Why is this (and the others like it) duplicated now? Is there a risk of getting out of sync?

Reply:
It's because there are two ways the process will be instantiated: either through reflection with the servlet which looks @ the annotations, or directly by either user code or via a unit test.  Direct instantiation won't address those annotation values, so it's better to assign them to get consistent behavior between unit test and runtime.  We can open a feature request to close that gap in the default init method if you'd like.

Reply:
ok. makes sense.